### PR TITLE
Turetske/parameter generation

### PR DESCRIPTION
**Pull request by turetske**
Implements parameter generation into structs with go generator based on the parameters.yaml file. Also wraps all calls to viper.Get* that are using generated params.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ and it shares the same defaults and behavior as stashcp.
 Building
 --------
 
+If the `parameters.yaml` file was altered, regenerate `parameters.go` by running `go generate` in the `./config` directory
```
Comment (bbockelm) [inline README.md:62]: Is this comment needed?  I believe `goreleaser` (and just hitting `make`) automatically runs `go generate`, no?

**Review by bbockelm — CHANGES_REQUESTED** [review]
I love the general approach here... exactly what I was thinking of.

However, two medium-sized changes:

1.  I don't like having this in the config module: I'm worried about conflicts between the parameter names and various configuration functions.  How about putting this into a standalone `param` module?
2. We shouldn't be putting the autogenerated file into the repo.  That's going to introduce unnecessary churn.  Instead, let's add the output file to `.gitignore` and remove it from the patch series.

File: README.md (hunk 1 above)
Comment (turetske) [reply README.md]: Removed from the code

**Comment by turetske** [comment]
@bbockelm 

I've updated with the changes you requested.

**Review by bbockelm — APPROVED** [review]
LGTM!  All the items from the prior review have been addressed.

**Pull request merged**